src/worker/ingestion/process-event.ts

```diff
@@ -299,10 +264,51 @@ export class EventsProcessor {
         return returnedProps
     }
 
+    private async setIsIdentified(teamId: number, distinctId: string, isIdentified = true): Promise<void> {
+        let personFound = await this.db.fetchPerson(teamId, distinctId)
+        if (!personFound) {
+            try {
+                personFound = await this.db.createPerson(
+                    DateTime.utc(),
+                    {},
+                    teamId,
+                    null,
+                    true,
+                    new UUIDT().toString(),
+                    [distinctId]
+                )
+            } catch {
+                // Catch race condition where in between getting and creating,
+                // another request already created this person
+                personFound = await this.db.fetchPerson(teamId, distinctId)
+            }
+        }
+        if (personFound && !personFound.is_identified) {
+            await this.db.updatePerson(personFound, { is_identified: isIdentified })
+        }
+    }
+
+    private async handleIdentifyOrAlias(
+        event: string,
+        properties: Properties,
+        distinctId: string,
+        teamId: number
+    ): Promise<void> {
+        if (event === '$create_alias') {
+            await this.alias(properties['alias'], distinctId, teamId, AliasTriggerEvent.CreateAlias)
+        } else if (event === '$identify') {
+            if (properties['$anon_distinct_id']) {
+                await this.alias(properties['$anon_distinct_id'], distinctId, teamId)
+            }
+            await this.setIsIdentified(teamId, distinctId)
+        }
+    }
+
     private async alias(
         previousDistinctId: string,
         distinctId: string,
         teamId: number,
+        aliasTriggerEvent = AliasTriggerEvent.Identify,
```

Comment:
It seems strange to require `alias` to understand how it was triggered. If we add another way to trigger it, does this mean we would have to update the enum and decide on the desired behaviour. Might it be better to just name it something like `shouldBeMergeIdentifiedPersons` ?

Reply:
Yup, that's valid. Originally had a boolean here but thought this was more explicit. Your comment makes sense though